### ISSUE TITLE
Exported nodes aren't available / set in [Tool] scripts

### ISSUE THREAD

**Issue by timshannon** [godotengine/godot#76182]
### Godot version

v4.0.2.stable.mono.official [7a0977ce2]

### System information

Ubuntu 22.10

### Issue description

Exported nodes don't seem to be set before a script marked as [Tool] is run in the editor. 

### Steps to reproduce

Either use the attached project or do the following:

1. Create a new scene, and attach a new C# script to the root node
```C#
using Godot;
using System;

[Tool]
public partial class main : Node3D
{
    [Export]
    public Node3D NodeTest;

    // Called every frame. 'delta' is the elapsed time since the previous frame.
    public override void _Process(double delta)
    {
        GD.Print(NodeTest.Name);
    }
}
```

2. Build the solution, and associate any node to that exported variable.

Even though it's set, the node isn't accessible from within the editor.

### Minimal reproduction project

[bug.zip](https://github.com/godotengine/godot/files/11255394/bug.zip)

**Comment by timshannon** [comment]
Confirmed, yes, it happens with GDScript as well:

![image](https://user-images.githubusercontent.com/1330485/232607132-411426aa-ad9c-4743-9ab5-0edeacabc6fa.png)

**Comment by KoBeWi** [comment]
Duplicate of #74141, unless other types are also affected?

**Comment by raulsntos** [comment]
- This seems like it was fixed by https://github.com/godotengine/godot/pull/76389. The PR is included in Godot 4.1.
- Can you still reproduce this issue in Godot 4.1 or any later release?

**Comment by timshannon** [comment]
@raulsntos confirmed working great now in 4.1.

Thanks,
